estate/models/estate_property_offer.py

```diff
@@ -15,3 +16,34 @@ class EstatePropertyOffer(models.Model):
     )
     partner_id = fields.Many2one("res.partner", required=True)
     property_id = fields.Many2one("estate.property", required=True)
+    validity = fields.Integer(default=7, string="Validity (days)")
+    date_deadline = fields.Date(
+        compute="_compute_date_deadline",
+        inverse="_inverse_date_deadline",
+        string="Deadline",
+    )
+
+    @api.depends("create_date", "validity")
+    def _compute_date_deadline(self):
+        for record in self:
+            if not record.create_date:
+                record.date_deadline = fields.Date.add(fields.Date.today(), days=record.validity)
+            else:
+                record.date_deadline = fields.Date.add(record.create_date.date(), days=record.validity)
```

Comment:
I leave you a suggestion to avoid using the `add()` method twice. You can apply it or leave the current code, I'll accept it anyway, just keep this type of suggestion in mind when you have to make calculated fields in real tasks, due to performance issues.

```suggestion
            date = record.create_date or fields.Date.today()
            record.date_deadline = fields.Date.add(date, days=record.validity)
```